### PR TITLE
Metadata for GHC 9.4.7

### PR DESCRIPTION
**Pull request by wz1000**

### CODE REVIEW

**Comment by wz1000** [comment]
https://gitlab.haskell.org/ghc/ghcup-ci/-/pipelines/83202

**Comment by hasufell** [comment]
The signatures are still invalid.

File: ghcup-0.0.7.yaml [hunk 1]
```diff
@@ -3494,6 +3494,100 @@ ghcupDownloads:
...
+              dlSubdir: ghc-9.4.7-x86_64-apple-darwin
+              dlHash: 2c874dc685cb72b0c4d6f226b795051705a923c25080eeba05d546350474cb1e
+          Windows:
+            unknown_versioning:
+              dlUri: https://downloads.haskell.org/~ghc/9.4.7/ghc-9.4.7-x86_64-unknown-mingw32.tar.xz
+              dlSubdir: ghc-9.4.7-x86_64-unknown-mingw32
+              dlHash: 415e6adebd764d278ecfe4ebf656ff96902953b8b669fba7dae2525af6665172
+          Linux_Alpine:
+            unknown_versioning:
+              dlUri: https://downloads.haskell.org/~ghc/9.4.7/ghc-9.4.7-x86_64-alpine3_12-linux.tar.xz
```
Comment (hasufell) [inline ghcup-0.0.7.yaml:3564]: Is this dynamic or static?

**Review by hasufell — COMMENTED** [review]
Please run the GHCup bindist pipeline. GHC CI pipelines are nit sufficient.

File: ghcup-0.0.7.yaml (hunk 1 above)
Comment (wz1000) [reply ghcup-0.0.7.yaml:3564]: This is dynamic now, as requested in https://gitlab.haskell.org/ghc/ghc/-/issues/23828

Static bindists are quite broken.

**Comment by wz1000** [comment]
Don't merge yet, I'm currently considering whether to backport an additional patch https://gitlab.haskell.org/ghc/ghc/-/issues/23819#note_522030

**Comment by wz1000** [comment]
@hasufell this is ready to go, we are not going to do any additional backports https://gitlab.haskell.org/ghc/ghc/-/issues/23819#note_522346

Could you merge this and sign with your keys? We can sort out the key situation later when Ben is also around.

**Comment by wz1000** [comment]
https://github.com/haskell/ghcup-metadata/actions/runs/5974734715

**Comment by hasufell** [comment]
I'm drunk. I'll look at it tomorrow. We need proper GPG signatures.

**Comment by hasufell** [comment]
Why did you merge with a broken and undocumented signature?

**Comment by wz1000** [comment]
> Why did you merge with a broken and undocumented signature?

Before I merged I asked @bgamari to fix the signatures and merge develop into master.

**Comment by hasufell** [comment]
Neither his or your signatures are documented. It's broken for end users.

**Comment by wz1000** [comment]
Ben's signature is documented at https://www.haskell.org/ghcup/guide/#gpg-verification

**Comment by hasufell** [comment]
> Ben's signature is documented at https://www.haskell.org/ghcup/guide/#gpg-verification

That is the old expired GPG key.

**Comment by bgamari** [comment]
I verified the signatures before merging. `for i in *.sig; do echo $i; gpg --verify $i || break; done` did not reveal any problematic signatures.

**Comment by bgamari** [comment]
> That is the old expired GPG key.

As far as I know that key should have been renewed several months ago:

$ gpg --list-keys FE5AB6C91FEA597C3B31180B73EDE9E8CFBAEF01
pub   rsa2048 2013-07-31 [SC] [expires: 2025-08-02]
      FE5AB6C91FEA597C3B31180B73EDE9E8CFBAEF01
uid           [ultimate] Benjamin Gamari <ben@smart-cactus.org>
uid           [ultimate] Benjamin Gamari <bgamari@gmail.com>
sub   rsa2048 2016-06-22 []
sub   rsa2048 2016-06-22 [E] [expires: 2025-08-02]
```

**Pull request merged**